bin/check_required_files_present

```diff
@@ -1,64 +1,64 @@
 #!/usr/bin/env bash
-MISSING_FILES=0
-SHOW_PROGRESS='false'
+missing_files=0
+show_progress='false'
 
-if [ ! -d exercises ] ; then
-	echo "\"exercises\" directory not found, make sure you're running from the repository root" >&2
-	exit 1
+if [[ ! -d exercises ]] ; then
+    echo '"exercises" directory not found, make sure you are running from the repository root' >&2
+    exit 1
 fi
 
 function show_progress {
-	if [ $SHOW_PROGRESS == 'true' ]; then
-		echo -n '.'
-	fi
+    if [[ "${show_progress}" == 'true' ]]; then
+        printf '.'
+    fi
 }
 
 function require_file {
-	filename=$1
-	if [ ! -f $filename ]; then
-		echo "required file missing: $filename" >&2
-		let "MISSING_FILES+=1"
-	fi
+    local filename=$1
+    if [[ ! -f "${filename}" ]]; then
+        echo "required file missing: $filename" >&2
+        missing_files=$(( missing_files + 1))
+    fi
 }
 
 function require_file_or_alternative {
-	filename=$1	
-	alternative_filename=$2
-	if [ ! -f $filename ] && [ ! -f $alternative_filename ]; then		
-		echo "required file missing: $filename or $alternative_filename" >&2
-		let "MISSING_FILES+=1"
-	fi
+    local filename=$1
+    local alternative_filename=$2
+    if [[ ! -f "${filename}" && ! -f "${alternative_filename}" ]]; then
+        echo "required file missing: ${filename} or ${alternative_filename}" >&2
+        missing_files=$(( missing_files + 1 ))
+    fi
 }
 
 function check_directory {
-	directory=$1
-	for exercise_directory in $directory/* ; do
-		show_progress
-		require_file_or_alternative "$exercise_directory/description.md" "$exercise_directory/instructions.md"
-		require_file "$exercise_directory/metadata.toml"
-	done
+    local directory=$1
+    for exercise_directory in "${directory}"/* ; do
+        show_progress
+        require_file_or_alternative "${exercise_directory}/description.md" "${exercise_directory}/instructions.md"
+        require_file "${exercise_directory}/metadata.toml"
+    done
 
 }
 
 while :
 do
-	case "$1" in
-		-p | --progress)
-			SHOW_PROGRESS="true"
-			shift 1
-			;;
-		*)
-			check_directory 'exercises'
-			if [ $SHOW_PROGRESS == 'true' ]; then
-				echo
-				echo Done: $MISSING_FILES files missing.
-			fi
-			if (( $MISSING_FILES > 0 )); then
-				exit 1
-			else
-				exit 0
-			fi
-			;;
-	esac
+    case "$1" in
+        -p | --progress)
+            show_progress="true"
+            shift 1
+            ;;
+        *)
+            check_directory 'exercises'
+            if [[ "${show_progress}" == 'true' ]]; then
+                echo
+                echo "Done: ${missing_files} files missing."
+            fi
+            if (( missing_files > 0 )); then
+                exit 1
+            else
+                exit 0
+            fi
```

Comment:
Is `exit "$(( missing_files == 0 ))"` too obscure?

Reply:
I'd prefer `(( missing_files == 0 )); exit` ... but being verbose here is probably good. It's longer but less likely to give anyone pause.